config/consensus_test.go

```diff
@@ -34,3 +36,20 @@ func TestConsensusParams(t *testing.T) {
 		}
 	}
 }
+
+// TestConsensusUpgradeWindow ensures that the upgrade window is a non-zero value, and confirm to be within the valid range.
+func TestConsensusUpgradeWindow(t *testing.T) {
+	for proto, params := range Consensus {
+		require.GreaterOrEqualf(t, params.MaxUpgradeWaitRounds, params.MinUpgradeWaitRounds, "Version :%v", proto)
+		for toVersion, delay := range params.ApprovedUpgrades {
+			if params.MinUpgradeWaitRounds != 0 || params.MaxUpgradeWaitRounds != 0 {
+				require.NotZerof(t, delay, "From :%v\nTo :%v", proto, toVersion)
+				require.GreaterOrEqualf(t, delay, params.MinUpgradeWaitRounds, "From :%v\nTo :%v", proto, toVersion)
+				require.LessOrEqualf(t, delay, params.MaxUpgradeWaitRounds, "From :%v\nTo :%v", proto, toVersion)
+			} else {
```

Comment:
Should this if condition check the MinUpgradeWaitRounds and MAxUpgradeWaitRounds independently?  For example, if MinUpgradeWaitRounds = 100 and MaxUpgradeWaitRounds = 0 , and delay = 200 will fail.

Reply:
wouldn't `require.GreaterOrEqualf(t, params.MaxUpgradeWaitRounds, params.MinUpgradeWaitRounds, "Version :%v", proto)` already cover that ?